### PR TITLE
Add validation and tests for VK_EXT_attachment_feedback_loop_layout

### PR DESCRIPTION
**Pull request by Tony-LunarG**
This will still get false positives for 04437/04439, but those have been removed from the spec and will be addressed by the solution to #4312

### CODE REVIEW

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -827,9 +866,15 @@ bool CoreChecks::ValidateBarriersToImages(const Location &outer_loc, const CMD_B
         if (!image_state) {
             continue;
         }
-        VkImageUsageFlags usage_flags = image_state->createInfo.usage;
-        skip |= ValidateBarrierLayoutToImageUsage(loc.dot(Field::oldLayout), img_barrier.image, img_barrier.oldLayout, usage_flags);
-        skip |= ValidateBarrierLayoutToImageUsage(loc.dot(Field::newLayout), img_barrier.image, img_barrier.newLayout, usage_flags);
+
+        if ((img_barrier.srcQueueFamilyIndex != img_barrier.dstQueueFamilyIndex) ||
```
Comment (jeremyg-lunarg) [inline layers/buffer_validation.cpp:870]: this if guard seems like an important fix even without the new layout & usage. Was not having it causing false positives in CTS?

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

File: layers/buffer_validation.cpp (hunk 1 above)
Comment (Tony-LunarG) [reply layers/buffer_validation.cpp:870]: I meant to mention that.  It looks like, at some point, all of the layout vs usage VUIDs (not just the new one) got prefaced with "If srcQueueFamilyIndex and dstQueueFamilyIndex define a queue family ownership transfer or oldLayout and newLayout define an image layout transition" so this was my attempt to implement that.  But I'm not at all sure that this is all there is to it and I'm open to suggestions.

File: layers/buffer_validation.cpp [hunk 2]
```diff
@@ -452,6 +452,41 @@ bool CoreChecks::ValidateRenderPassLayoutAgainstFramebufferImageUsage(RenderPass
...
+        if (((image_usage & (VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT | VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT)) == 0) ||
+            ((image_usage & (VK_IMAGE_USAGE_SAMPLED_BIT | VK_IMAGE_USAGE_INPUT_ATTACHMENT_BIT)) == 0)) {
+            vuid = use_rp2 ? "VUID-vkCmdBeginRenderPass2-initialLayout-07002" : "VUID-vkCmdBeginRenderPass-initialLayout-07000";
+            LogObjectList objlist(image);
+            objlist.add(renderpass);
+            objlist.add(framebuffer);
+            objlist.add(image_view);
+            skip |=
+                LogError(objlist, vuid,
+                         "%s: Layout/usage mismatch for attachment %u in %s"
```
Comment (ncesario-lunarg) [inline layers/buffer_validation.cpp:465]: `%u` -> `PRIu32` (same for below)?

Every time I see this I'm tempted to stop everything and refactor the logging :'(

File: layers/buffer_validation.cpp (hunk 1 above)
Comment (jeremyg-lunarg) [reply layers/buffer_validation.cpp:870]: It looks like you got it right to me.  Looks like it came in here https://github.com/KhronosGroup/Vulkan-Docs/commit/dd7b521af03ed3ce13b7d2a54c4542f5af7cf370  (see last bullet point in the commit message).  I think that this was done to get ready for synchronization2.

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp [hunk 3]
```diff
@@ -2263,17 +2263,25 @@ TEST_F(VkLayerTest, RenderPassBeginLayoutsFramebufferImageUsageMismatches) {
...
         maintenance2Supported = true;
     }
 
-    ASSERT_NO_FATAL_FAILURE(InitState(nullptr, nullptr, VK_COMMAND_POOL_CREATE_RESET_COMMAND_BUFFER_BIT));
+    auto attachment_feedback_loop_layout_features =
+        LvlInitStruct<VkPhysicalDeviceAttachmentFeedbackLoopLayoutFeaturesEXT>();
+    auto features2 = LvlInitStruct<VkPhysicalDeviceFeatures2>(&attachment_feedback_loop_layout_features);
+    auto vkGetPhysicalDeviceFeatures2KHR = reinterpret_cast<PFN_vkGetPhysicalDeviceFeatures2KHR>(
+        vk::GetInstanceProcAddr(instance(), "vkGetPhysicalDeviceFeatures2KHR"));
+    vkGetPhysicalDeviceFeatures2KHR(gpu(), &features2);
```
Comment (paul-lunarg) [inline tests/vklayertests_descriptor_renderpass_framebuffer.cpp:2283]: You could use VkLayerTest::GetPhysicalDeviceFeatures2 instead, which does exactly this and also checks for Vk1.1

**Review by paul-lunarg — APPROVED** [review]

File: layers/buffer_validation.cpp (hunk 2 above)
Comment (Tony-LunarG) [reply layers/buffer_validation.cpp]: Ugh - I've been trying not to do that.  Really!  Fixed

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp (hunk 3 above)
Comment (Tony-LunarG) [reply tests/vklayertests_descriptor_renderpass_framebuffer.cpp]: Good idea - I forgot about that.  Fixed

**Pull request merged**